segway/output.py

```diff
@@ -30,24 +34,59 @@ def make_bed_attrs(mapping):
     return "track %s" % res
 
 
-# Takes a list of filepaths, each of which points to a segmentation bed file.
-# Concatenates the files, merging entries if necessary.  Start file with
-# header.  Used by IdentifySaver and PosteriorSaver.
-def concatenate_window_segmentations(window_filenames, header, outfilename):
+def create_color_assignment(window_filenames):
+    # Build a color assignment for each superlabel in the window files
+    labels = set()
+    for window_filename in window_filenames:
+        with maybe_gzip_open(window_filename) as window_file:
+            for line in window_file:
+                label = extract_superlabel(line.split()[LABEL_INDEX])
+                labels.add(label)
+    # If labels are numbers, use as indices into color list
+    if all(label.isdecimal() for label in labels): # Index into color list
+        return {label: ','.join(map(str, SCHEME[int(label) % NUM_COLORS]))
+                for label in labels}
+    # Otherwise, assign colors by alphabetical order
+    return {label: ','.join(map(str, SCHEME[i % NUM_COLORS])) 
+            for i, label in enumerate(sorted(list(labels)))}
+
+
+# Takes a list of filepaths, each of which points to a segmentation BED4 file.
+# Concatenates the files, merging entries if necessary. Produces a BED file 
+# and a track file with an additional header, by default with 9 columns. 
+# Used by IdentifySaver and PosteriorSaver.
+def merge_windows_to_bed(window_filenames, header, bedfilename, trackfilename,
+                         as9cols = True):
```

Comment:
at the very least this should probably be `as_9_cols`. Perhaps `bed9_output`? It is tricky to name I will admit.